vnext/JSI/Shared/ChakraRuntime.cpp

```diff
@@ -589,354 +567,246 @@ facebook::jsi::Runtime::ScopeState *ChakraRuntime::pushScope() {
 
 void ChakraRuntime::popScope([[maybe_unused]] Runtime::ScopeState *state) {
   assert(state == nullptr);
-  VerifyJsErrorElseThrow(JsCollectGarbage(m_runtime));
+  ChakraVerifyJsErrorElseThrow(JsCollectGarbage(m_runtime));
 }
 
 bool ChakraRuntime::strictEquals(const facebook::jsi::Symbol &a, const facebook::jsi::Symbol &b) const {
-  return CompareJsValues(GetChakraObjectRef(a), GetChakraObjectRef(b));
+  return StrictEquals(GetJsRef(a), GetJsRef(b));
 }
 
 bool ChakraRuntime::strictEquals(const facebook::jsi::String &a, const facebook::jsi::String &b) const {
-  return CompareJsValues(GetChakraObjectRef(a), GetChakraObjectRef(b));
+  return StrictEquals(GetJsRef(a), GetJsRef(b));
 }
 
 bool ChakraRuntime::strictEquals(const facebook::jsi::Object &a, const facebook::jsi::Object &b) const {
-  return CompareJsValues(GetChakraObjectRef(a), GetChakraObjectRef(b));
+  return StrictEquals(GetJsRef(a), GetJsRef(b));
 }
 
 bool ChakraRuntime::instanceOf(const facebook::jsi::Object &obj, const facebook::jsi::Function &func) {
-  bool result;
-  VerifyJsErrorElseThrow(JsInstanceOf(GetChakraObjectRef(obj), GetChakraObjectRef(func), &result));
-  return result;
+  return InstanceOf(GetJsRef(obj), GetJsRef(func));
 }
 
 #pragma endregion Functions_inherited_from_Runtime
 
-void ChakraRuntime::VerifyJsErrorElseThrow(JsErrorCode error) {
-  switch (error) {
-    case JsNoError: {
-      return;
-      break;
-    }
+[[noreturn]] void ChakraRuntime::ThrowJsExceptionOverride(JsErrorCode errorCode, JsValueRef jsError) {
+  if (errorCode == JsErrorScriptException || GetValueType(jsError) == JsError) {
+    RewriteErrorMessage(jsError);
+    throw facebook::jsi::JSError(*this, ToJsiValue(jsError));
+  } else {
+    std::ostringstream errorString;
+    errorString << "A call to Chakra API returned error code 0x" << std::hex << errorCode << '.';
```

Comment:
>std::hex << errorCode [](start = 68, length = 21)

Should this be in parenthesis? Or is `<<`  right associative?

Reply:
This is a correct syntax. It is left associate. To format number as hex we must first put std::hex into the stream. I think it changes the number format mode.

---
In reply to: [470111422](https://github.com/microsoft/react-native-windows/pull/5710#discussion_r470111422) [](ancestors = 470111422)